engine/src/main/java/dgca/verifier/app/engine/DefaultCertLogicEngine.kt

```diff
@@ -67,48 +67,66 @@ class DefaultCertLogicEngine(
-                val res = when {
-                    rule.engine == CERTLOGIC_KEY && ruleEngineVersion != null && CERTLOGIC_VERSION.isGreaterOrEqualThan(
-                        ruleEngineVersion
-                    ) && hcertVersion != null && schemaVersion != null && hcertVersion.first == schemaVersion.first && hcertVersion.isGreaterOrEqualThan(
-                        schemaVersion
-                    ) ->
-                        when (jsonLogicValidator.isDataValid(
-                            rule.logic,
-                            dataJsonNode
-                        )) {
-                            true -> Result.PASSED
-                            false -> Result.FAIL
-                            else -> Result.OPEN
-                        }
-                    else -> Result.OPEN
+        if (rules.isEmpty()) return emptyList()
+
+        val dataJsonNode = prepareData(externalParameter, payload)
+        val hcertVersion: Triple<Int, Int, Int>? = hcertVersionString.toVersion()
+
+        return rules.map {
+            checkRule(
+                rule = it,
+                dataJsonNode = dataJsonNode,
+                hcertVersion = hcertVersion,
+                certificateType = certificateType
+            )
+        }
+    }
+
+    private fun checkRule(
+        rule: Rule,
+        dataJsonNode: ObjectNode,
+        hcertVersion: Triple<Int, Int, Int>?,
+        certificateType: CertificateType
+    ): ValidationResult {
+        val ruleEngineVersion = rule.engineVersion.toVersion()
+        val schemaVersion = rule.schemaVersion.toVersion()
+
+        val validationErrors = mutableListOf<Exception>()
+
+        val isCompatibleVersion = rule.engine == CERTLOGIC_KEY &&
+                ruleEngineVersion != null &&
+                CERTLOGIC_VERSION.isGreaterOrEqualThan(ruleEngineVersion) &&
+                hcertVersion != null &&
+                schemaVersion != null &&
+                hcertVersion.first == schemaVersion.first &&
+                hcertVersion.isGreaterOrEqualThan(schemaVersion)
+
+        val res = if (isCompatibleVersion) {
+            try {
+                when (jsonLogicValidator.isDataValid(rule.logic, dataJsonNode)) {
+                    true -> Result.PASSED
+                    false -> Result.FAIL
                 }
-                val cur: String = affectedFieldsDataRetriever.getAffectedFieldsData(
-                    rule,
-                    dataJsonNode,
-                    certificateType
-                )
-                validationResults.add(
-                    ValidationResult(
-                        rule,
-                        res,
-                        cur,
-                        null
-                    )
-                )
+            } catch (e: Exception) {
+                validationErrors.add(e)
+                Result.OPEN
             }
-            validationResults
         } else {
-            emptyList()
+            Result.OPEN
         }
```

Comment:
Still map exceptions to `OPEN`, but store the `Exception` for the result.

`Exception` instead of `Throwable` to not catch `Error`.
https://stackoverflow.com/questions/6083248/is-it-a-bad-practice-to-catch-throwable